geostacks/geostacks.py

```diff
@@ -95,29 +145,24 @@ def query_pathrow(self, point_geometry):
         '''
 
         # (1) Ball Tree
-        points = np.vstack((self.footprint.lon_CTR.values,
-                            self.footprint.lat_CTR.values)).T
-        points *= np.pi/180.
-        LSBall = BallTree(points, metric='haversine')
-
-        q = np.array(point_geometry)
-        q *= np.pi/180.
-        if type(point_geometry) is Point:
-            pt = point_geometry
-        else:
-            pt = Point(point_geometry)
+        # Load tree to reduce runtime...
+        data = pkgutil.get_data(__name__, "sensors/ls8Ball.pkl")
+        LSBall = pickle.loads(data)
+
+        q = self.make_pts(lat, lon)
 
-        pre_selection = LSBall.query_radius(q.reshape(1, -1),
-                                            r=0.05,
+        pre_selection = LSBall.query_radius(q, r=0.05,
                                             return_distance=False)
         pre_selection_idx = pre_selection[0]
+        # @whyjz : just curious, why the sort?
```

Comment:
This sort the indexes so that when we `print(ls8_index.corner_pts_df.iloc[idxs])`, the results are shown in an ascending order of path/row.